### PR TITLE
Change tsapi Au test to test TS API calls TSNetConnect() and TSNetAccept().

### PR DESCRIPTION
**Pull request by ywkaras**
This PR is dependent on https://github.com/apache/trafficserver/pull/6291 .

This is the first step in converting the tests in InkAPITest.cc (using a "synthetic" client/server) into Au tests.  The synthetic server / client use TSNetAccept() and TSNetConnect() respectively.  This is a dedicated test to exercise these two API functions.

### CODE REVIEW

File: plugins/xdebug/xdebug.cc [hunk 1]
```diff
@@ -31,10 +31,10 @@
 #include <unistd.h>
 
 #include <ts/ts.h>
-#include "tscore/ink_defs.h"
-#include "tscpp/util/PostScript.h"
-#include "tscpp/util/TextView.h"
-#include "Cleanup.h"
+#include <tscore/ink_defs.h>
```
Comment (randall) [inline plugins/xdebug/xdebug.cc:34]: This looks unrelated to the autest

File: iocore/net/UnixNetProcessor.cc [hunk 2]
```diff
@@ -237,16 +237,18 @@ UnixNetProcessor::connect_re_internal(Continuation *cont, sockaddr const *target
...
-  if (lock.is_locked()) {
-    MUTEX_TRY_LOCK(lock2, get_NetHandler(t)->mutex, t);
-    if (lock2.is_locked()) {
-      int ret;
-      ret = vc->connectUp(t, NO_FD);
-      if ((using_socks) && (ret == CONNECT_SUCCESS)) {
-        return &socksEntry->action_;
-      } else {
-        return ACTION_RESULT_DONE;
+  if (reinterpret_cast<EThread *>(this_thread()) == t) {
```
Comment (randall) [inline iocore/net/UnixNetProcessor.cc:240]: Is this supposed to be part of this change?

Comment (ywkaras) [reply iocore/net/UnixNetProcessor.cc:240]: Yes, the test will not work without this change @duke8253

File: plugins/xdebug/xdebug.cc (hunk 1 above)
Comment (ywkaras) [reply plugins/xdebug/xdebug.cc:34]: Yes and no.  The new plugin needs to use Cleanup.h, so the PR moves it to include/cpp/util.  <> seems more correct to me than "", because "" tell the compiler to look for the header file in the same directory as the including file first before looking for it in standard locations or directories specified with -I on the compile command line.  Cleaning that up is not strictly related to this PR.

**Comment by ywkaras** [comment]
[approve ci autest]

**Comment by ywkaras** [comment]
[approve ci Ubuntu]

**Comment by ywkaras** [comment]
[approve ci autest]

**Comment by ywkaras** [comment]
[approve ci autest]

**Comment by ywkaras** [comment]
[approve ci autest]

**Comment by ywkaras** [comment]
[approve ci autest]

File: iocore/net/UnixNetProcessor.cc (hunk 2 above)
Comment (randall) [reply iocore/net/UnixNetProcessor.cc:240]: I guess it seems like burying the lede here. The subject of the PR just mentions tests, but it looks like a general change in behavior here.

File: plugins/xdebug/xdebug.cc (hunk 1 above)
Comment (randall) [reply plugins/xdebug/xdebug.cc:34]: also `<` tends to indicate a system include file. Plus all of the other inclusions of ink_defs. in plugins use `"`

File: iocore/net/UnixNetProcessor.cc (hunk 2 above)
Comment (ywkaras) [reply iocore/net/UnixNetProcessor.cc:240]: So specifically are you asking me to change the commit comment?

**Comment by ywkaras** [comment]
I'm going to close this and pull these changes into https://github.com/apache/trafficserver/pull/6832 .
